discovery/gossiper.go

```diff
@@ -1634,8 +1634,18 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 			return nil
 		}
 
-		// Node announcement was successfully proceeded and know it
-		// might be broadcast to other connected nodes.
+		// In order to ensure we don't leak unadvertised nodes, we'll
+		// make a quick check to ensure this node intends to publicly
+		// advertise itself to the network.
+		isPublic, err := d.cfg.Router.IsPublicNode(node.PubKeyBytes)
+
+		// If it doesn't, we'll skip broadcasting this announcement to
+		// the rest of our peers.
+		if !isPublic || err != nil {
+			nMsg.err <- nil
+			return announcements
```

Comment:
I was thinking of the case where we receive a node announcement, and later we learn about a public channel for the node. E.g. we open a private channel with the node, receive a node announcement, then they open a public channel which we learn about. I _thiiink_ it's okay now, as they probably should send out a `NodeAnnouncement` after they open the public channel.